lusidtools/cocoon/utilities.py

```diff
@@ -1361,7 +1363,7 @@ def generate_time_based_unique_id(time_generator: None):
     timestamp = hex(int(current_time * 10000000.0))
     # Create the scope id by joining the hex representation with dashes every 4 characters
     uid = "-".join(timestamp[i : i + 4] for i in range(2, len(timestamp), 4))
-    return uid
+    return uid + str(uuid.uuid4())
```

Comment:
Not sure on whether this actually makes the id any more unique, let's verify there is an issue with the existing implementation then go from there.

Reply:
Verified, that this is an issue. Can you please move creating the uuid and appending to the create_scope_id function?